### PR TITLE
Only propagate info if the quantity stays effectively the same

### LINKED ISSUE #5047: When should `info` be linked to a new object?
Mostly for @taldcroft - I noticed that in `Quantity` the way we have set up `__array_finalize__`, `info` is passed on not just for views (where it should be), but also for copies (implicitly in arithmetic operations, etc.). Which ones are reasonable?  Just thinking about whether, e.g., `info.name` should be propagated, I'd think:

- Surely for
   * views & reshapes: `q[...]`, `q.squeeze`, etc.
   * insertions: `q.insert(...)`

- Probably for
   * selection of scalars: `q[0]` or in `for q1 in q:` (for columns this returns a scalar without `info`)
   * copies: `q.copy()` and equivalents
   * equivalent unit changes: `q.to(...)`, `q.si`, `q.decompose()`, etc.

- Probably not for
   * operations `q3 = q1 + q2`
   * real unit changes `q * unit` (including in-place??; `q /= u.m`)

What do you think?

p.s. Currently, all of the above happen, in part because I use `__array_finalize__` in `Quantity._new_view`, something which I don't think we had really considered when we made the change in `__array_finalize__`. But that also means that in principle it may not too hard to fairly finely define the behaviour.

### PR DESCRIPTION
**Pull request by mhvk**
### Description

This pull request is to address the fact that `info` gets copied from one quantity to another even when it makes little sense (such as in operations on quantities), slowing things done needlessly and possibly leading to confusion.

Fixes #5047, #14248

Also addresses https://github.com/astropy/astropy/issues/14244#issuecomment-1369524044

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/units/tests/test_quantity_info.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        q = self.q.copy()
+        q.insert(1, 1 * u.cm / u.hr)
+        assert_info_equal(q, self.q)
+
+    def test_unary_op(self):
+        q = -self.q
+        assert_no_info(q)
+
+    def test_binary_op(self):
+        q = self.q + self.q
```
Comment (taldcroft) [inline astropy/units/tests/test_quantity_info.py:86]: Do you need to test `-`, `*`, `/` here as well?

File: astropy/units/quantity.py [hunk 2]
```diff
@@ -1252,14 +1266,16 @@ def __rtruediv__(self, other):
         """Right Division between `Quantity` objects and other objects."""
 
         if isinstance(other, (UnitBase, str)):
-            return self._new_view(1.0 / self.value, other / self.unit)
+            return self._new_view(1.0 / self.value, other / self.unit, finalize=False)
```
Comment (taldcroft) [inline astropy/units/quantity.py:1269]: This change is not covered in tests.

File: astropy/units/tests/test_quantity_info.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Test the propagaion of info on Quantity during operations."""
```
Comment (taldcroft) [inline astropy/units/tests/test_quantity_info.py:2]: Typo.

**Review by taldcroft — COMMENTED** [review]
This looks generally good with some small comments.

My only big-picture question is about performance. How much improvement does this bring for the covered cases? Conversely, are there any possible performance regressions in other areas?

**Comment by mhvk** [comment]
@taldcroft - on the performance, it is quite an improvement:

```
In [2]: q = [2,4]*u.percent

In [3]: q2 = 1*u.percent

In [4]: %timeit q + q2
8.27 µs ± 101 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
--> reduced a little to 7.47 µs ± 71.8 ns
In [5]: q.info.name = 'q'

In [6]: %timeit q + q2
13 µs ± 153 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
--> reduced quite a bit to 7.62 µs ± 103 ns
```

File: astropy/units/tests/test_quantity_info.py (hunk 3 above)
Comment (mhvk) [reply astropy/units/tests/test_quantity_info.py]: fixed, thanks

File: astropy/units/tests/test_quantity_info.py (hunk 1 above)
Comment (taldcroft) [reply astropy/units/tests/test_quantity_info.py:86]: OK.

**Review by taldcroft — APPROVED** [review]
OK, looks good, *pending CI passing*. Once that happens feel free check the CI box in the description and merge.

**Pull request merged**